libs/movie/src/lib/movie/marketplace/main/main.component.ts

```diff
@@ -32,16 +32,20 @@ export class MainComponent implements OnInit {
     this.dynTitle.setPageTitle('Film Page', 'Main Info');
   }
 
-  showSalesPitch(movie: Movie) {
-    if (!movie.promotional?.videos?.salesPitch) return false;
-    const { privacy, jwPlayerId, description } = movie.promotional.videos.salesPitch;
-    return privacy === 'public' && (jwPlayerId || description);
-  }
-
   videoStateChanged(title: Movie, event: string) {
     if (event === 'play' && !this.alreadyPlayed) {
       this.analytics.addTitle('promoElementOpened', title);
       this.alreadyPlayed = true;
     }
   }
+
+  getMainVideo(movie: Movie) {
+    const video = movie.promotional.videos;
+    const otherVideo = video.otherVideos.filter(video => video.privacy === 'public' && video.storagePath !== '' && video.jwPlayerId !== '')
+    if (otherVideo.length) return otherVideo[0];
+    if (video.salesPitch?.storagePath && video.salesPitch?.privacy === 'public' && video.salesPitch?.jwPlayerId) return video.salesPitch;
+    if (video.publicScreener?.storagePath && video.publicScreener?.privacy === 'public' && video.publicScreener?.jwPlayerId) return video.publicScreener;
+    if (video.screener?.storagePath && video.screener?.jwPlayerId) return video.screener;
+  }
```

Comment:
Sales Pitch should be first as mentioned in the issue.
publicScreener is always public, no need to check but can be displayed on catalog app only
screener cannot be public, no need here.

```suggestion
  getMainVideo(movie: Movie) {
    const videos = movie.promotional.videos;
    const isPublicVideo = (video: MovieVideo) => {
      return video?.storagePath && video?.privacy === 'public' && video?.jwPlayerId;
    }

    if (isPublicVideo(videos.salesPitch)) return videos.salesPitch;
    const otherVideo = videos.otherVideos.find(video => isPublicVideo(video));
    if (otherVideo) return otherVideo;
    if (this.shell.currentApp === 'catalog' && isPublicVideo(videos.publicScreener)) return videos.publicScreener;
  }
```